provision/certs/certs.go

```diff
@@ -0,0 +1,145 @@
+package certs
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/mainflux/mainflux/errors"
+	mfSDK "github.com/mainflux/mainflux/sdk/go"
+)
+
+// Thing is Mainflux SDK thing.
+type Thing mfSDK.Thing
+
+// Channel is Mainflux SDK channel.
+type Channel mfSDK.Channel
+
+var (
+	// ErrCerts indicates error fetching certificates.
+	ErrCerts = errors.New("failed to fetch certs data")
+
+	// ErrCertsRemove indicates failure while cleaning up from the Certs service.
+	ErrCertsRemove = errors.New("failed to remove certificate")
+
+	// ErrConflict indicates duplicate unique field.
+	ErrConflict = errors.New("duplicate unique field")
+
+	// ErrUnauthorized indicates forbidden access.
+	ErrUnauthorized = errors.New("unauthorized access")
+
+	// ErrMalformedEntity indicates malformed request data.
+	ErrMalformedEntity = errors.New("malformed data")
+
+	// ErrNotFound indicates that entity doesn't exist.
+	ErrNotFound = errors.New("entity not found")
+)
+
+// BSConfig represents Config entity to be stored by Bootstrap service.
+type BSConfig struct {
+	ThingID     string   `json:"thing_id,omitempty"`
+	ExternalID  string   `json:"external_id,omitempty"`
+	ExternalKey string   `json:"external_key,omitempty"`
+	Channels    []string `json:"channels,omitempty"`
+	Content     string   `json:"content,omitempty"`
+	ClientCert  string   `json:"client_cert,omitempty"`
+	ClientKey   string   `json:"client_key,omitempty"`
+	CACert      string   `json:"ca_cert,omitempty"`
+}
+
+// Cert represents certs data.
+type Cert struct {
+	CACert     string `json:"ca_cert,omitempty"`
+	ClientKey  string `json:"client_key,omitempty"`
+	ClientCert string `json:"client_cert,omitempty"`
+}
+
+// SDK is wrapper around cert SDK that adds certs management.
+type SDK interface {
+
+	// Cert creates cert using external PKI provider and Certs service.
+	Cert(thingID, thingKey, token string) (Cert, error)
+
+	// RemoveCert revokes and removes cert from Certs service.
+	RemoveCert(key, token string) error
+}
+
+type provisionSDK struct {
```

Comment:
This is supposed support certificate service

Reply:
I mean - do we need this wraper, or we can extend our official SDK to contain these new functions (that you produced by wrapping)?

Reply:
We don't have cert service in sdk

Reply:
And here we have? Or you are mocking it? And if you are mocking it here, why not mocking it directly in SDK, where it should be?

Reply:
well not quite mocking, if you provide and url and response from that url matches expected that it could work
maybe, but than I would create another issue for that and have a discussion on a topic